core/config/config.go

```diff
@@ -61,15 +61,31 @@ type IncredibleSquaringContractsRaw struct {
 	OperatorStateRetrieverAddr string `json:"operatorStateRetriever"`
 }
 
-// NewConfig parses config file to read from from flags or environment variables
+// NewConfig parses config file to read from flags or environment variables
 // Note: This config is shared by challenger and aggregator and so we put in the core.
 // Operator has a different config and is meant to be used by the operator CLI.
 func NewConfig(ctx *cli.Context) (*Config, error) {
 
 	var configRaw ConfigRaw
-	configFilePath := ctx.GlobalString(ConfigFileFlag.Name)
-	if configFilePath != "" {
-		sdkutils.ReadYamlConfig(configFilePath, &configRaw)
+
+	environment := ctx.GlobalString(EnvironmentFlag.Name)
+	if environment != "" {
+		configRaw.Environment = sdklogging.LogLevel(environment)
+	}
+
+	ethRpcUrl := ctx.GlobalString(EthRpcUrlFlag.Name)
+	if ethRpcUrl != "" {
+		configRaw.EthRpcUrl = ethRpcUrl
+	}
```

Comment:
these feel like bugs. What if the ethRpcUrl is set to ""? You should probably return an error.